src/main/java/org/sunbird/digilocker/service/DigiLockerIntegrationServiceImpl.java

```diff
@@ -0,0 +1,243 @@
+
+    @Autowired
+    UserUtilityService userUtilityService;
+
+    @Autowired
+    CassandraOperation cassandraOperation;
+
+    @Autowired
+    OutboundRequestHandlerServiceImpl outboundRequestHandlerService;
+
+    @Autowired
+    CbExtServerProperties serverProperties;
+
+    private Logger logger = LoggerFactory.getLogger(getClass().getName());
+    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+
+    @Override
+    public PullURIResponse generateURIResponse(PullURIRequest request) {
+        PullURIResponse response = new PullURIResponse();
+        Map<String, Object> getUserInfo = userUtilityService.getUserDetails(Constants.PHONE, request.getDocDetails().getMobile());
+        String certificateAccessCode = request.getDocDetails().getCertificateAccessCode();
+        if (MapUtils.isNotEmpty(getUserInfo)) {
+            String userId = (String) getUserInfo.get(Constants.IDENTIFIER);
+            if (StringUtils.isNotBlank(userId)) {
+                Map<String, Object> userInfo = new HashMap<>();
+                userInfo.put(Constants.USER_ID, userId);
+                List<Map<String, Object>> userEnrollmentInfo = cassandraOperation.getRecordsByPropertiesWithoutFiltering(
+                        Constants.KEYSPACE_SUNBIRD_COURSES, Constants.TABLE_USER_ENROLMENT, userInfo, Arrays.asList(Constants.USER_ID, Constants.COURSE_ID, Constants.ISSUED_CERTIFICATES));
+                if (CollectionUtils.isNotEmpty(userEnrollmentInfo)) {
+                    userEnrollmentInfo = userEnrollmentInfo.stream()
+                            .filter(enrollment -> {
+                                List<Map<String, Object>> issuedCertificates = (List<Map<String, Object>>) enrollment.get(Constants.ISSUED_USER_CERTIFICATE);
+                                return CollectionUtils.isNotEmpty(issuedCertificates) &&
+                                        issuedCertificates.stream()
+                                                .map(cert -> certificateAccessCode.equalsIgnoreCase((String) cert.get("token")))
+                                                .anyMatch(Boolean.TRUE::equals);
+                            }).collect(Collectors.toList());
+                    if ((CollectionUtils.isEmpty(userEnrollmentInfo)) || (CollectionUtils.isNotEmpty(userEnrollmentInfo) && userEnrollmentInfo.size() > 1)) {
+                        response.setResponseStatus(new PullURIResponse.ResponseStatus("0", new Date().toString(), request.getTxn()));
+                        return response;
+                    } else {
+                        PullURIResponse.DocDetails docDetails = new PullURIResponse.DocDetails();
+                        PullURIResponse.IssuedTo issuedTo = new PullURIResponse.IssuedTo();
+                        PullURIResponse.Persons persons = new PullURIResponse.Persons();
+                        PullURIResponse.Person person = new PullURIResponse.Person();
```

Comment:
These should be inside the PullURIResponse itself... 
Upon creating PullURIResponse object -- it should create one instance of all the children object it requires... 
This way it is easier to understand and easier to read the code...